server/db/driver/pg/internal/matches.go

```diff
@@ -126,15 +126,17 @@ const (
 	CompletedOrAtFaultMatchesLastN = `
 		WITH acct (aid) AS ( VALUES($1::BYTEA) )
 
-		SELECT status, (status=4 OR (status=3 AND makerAccount = aid)) AS success,
+		SELECT status, (status=4 OR (status=3 AND makerAccount = aid AND takerAccount != aid)) AS success,
```

Comment:
Neglected the maker==taker situation where we don't count MakerRedeemed as success if the user is also the taker.